subsys/bluetooth/host/l2cap.c

```diff
@@ -483,8 +502,8 @@ struct net_buf *bt_l2cap_create_pdu_timeout(struct net_buf_pool *pool,
 					  timeout);
 }
 
-void bt_l2cap_send_cb(struct bt_conn *conn, u16_t cid, struct net_buf *buf,
-		      bt_conn_tx_cb_t cb, void *user_data)
+int bt_l2cap_send_cb(struct bt_conn *conn, u16_t cid, struct net_buf *buf,
```

Comment:
bt_l2cap_send is just a macro os it is already updated by this patch. Regarding the ownership nothing has changed, the buffer passed on this are owned by the stack, so it works exactly like bt_conn_send_cb, so if bt_conn_send_cb fails it unref the buffer, that is it. The same rule also applies to bt_l2cap_chan_send, but since it actually queues packets failing to sending a segment won't drop the packets, they are just processed later.

Reply:
No, bt_l2cap_send is not a macro, it is an inline function, which ignores the return value of bt_l2cap_send_cb.

Yes, the ownership is still the same, but I think it would be beneficial to document this anyway. Especially now that we are ignoring the return value in multiple places. The `net_buf_unref` handling is a bit all over the place, so it would be nice with some comments stating the current behavior.

Reply:
Ive fixed that now, it should have a return now.

Reply:
I mean bt_l2cap_send does never fail, there are comments regarding this behavior on the ATT PR.

Reply:
Yeah, I just meant that even without a callback `bt_conn_send_cb` could still fail here:
https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/host/conn.c#L1254

I think a comment with the assumptions about why it is safe to ignore the return value would be beneficial.